pkg/provider/azure_controller_common.go

```diff
@@ -663,3 +711,25 @@ func isInstanceNotFoundError(err error) bool {
 	}
 	return strings.Contains(errMsg, errStatusCode400) && strings.Contains(errMsg, errInvalidParameter) && strings.Contains(errMsg, errTargetInstanceIds)
 }
+
+func getErrorCode(err error) *string {
+	if err == nil {
+		return nil
+	}
+	matches := errorCodeRE.FindStringSubmatch(err.Error())
+	if matches == nil {
+		return nil
+	}
+	return &matches[1]
+}
+
+type FutureParser interface {
+	ConfigAccepted(future *azure.Future) bool
+}
+
+type controllerFutureParser struct{}
+
+func (c *controllerFutureParser) ConfigAccepted(future *azure.Future) bool {
```

Comment:
I think write a simple function `ConfigAccepted(future *azure.Future) bool` would be more clear, and add comments for this function.